#533 Feature: Support Configuration Store values to use properties to inject values (#541)
boozallen/aissemble: docs/modules/ROOT/pages/configuration-store.adoc

NOTE: The `topic` property is overwritten when the environment specific value `ciTopic`.

> The Configuration Store also support simplified way to provide config file instead of creating additional .properties file.

Reviewer: ```suggestion
The Configuration Store also support simplified way to provide configurations instead of mounting properties files.
```

> The Configuration Store also support simplified way to provide config file instead of creating additional .properties file.

Reviewer: This should be moved to the bottom of the "Mounting Properties Files" section of this doc

> Under values.yaml, specifying base and environment specific properties can act as configuration properties file.

Reviewer: ```suggestion
Within the Helm chart, one can directly define properties files with their included configurations.
```

Then we should hyperlink part of the above text "define properties files" to the new section in the helm chart readme.

[source,properties]
----
configMap:
  name: configuration-store-quarkus-config
  baseSparkProperties:
    - connector=smallrye-kafka
    - topic=baseTopic
  envSparkProperties:
    - connector=smallrye-kafka
    - auth-token=auth-secret

> ----

Reviewer: This should be moved to the helm chart README. Could create a new section title "Defining Properties Files"

NOTE: Multiple configuration properties files can be stored at either location.

> The Configuration Store also support simplified way to provide configurations instead of mounting properties files.

Reviewer 2: A: I think we need to make the choices here more obvious. It might make sense use separate sections. Maybe something like:
```adoc
==== Properties Files from Config Maps
The simplest way to mount properties files onto the Configuration Store deployment 
is to use ConfigMaps.  To support this case, the configuration store Helm chart 
supports [directly specifying base and environment properties] under values.yaml.

==== Properties Files from Volumes
_existing content_
```

Reviewer: Building on this:

```adoc
==== Properties Files from Config Maps
The simplest way to mount properties files onto the Configuration Store deployment 
is to use ConfigMaps.  To support this case, one can provide configurations by [defining properties files]
as part of the Helm chart's values.
```

Like to avoid using `values.yaml` cause it could also be there `values-dev.yaml` or `values-ci.yaml` so best to keep it generic.

Reviewer: Also when changing the header names, make sure to update the hyperlinks in the above `Deploying the Configuration Store` section

> The Configuration Store also support simplified way to provide configurations instead of mounting properties files.

Reviewer: ```suggestion
The Configuration Store also supports a simplified way of providing configurations without mounting properties files.
```

> Under values.yaml, https://github.com/boozallen/aissemble/blob/dev/extensions/extensions-helm/aissemble-configuration-store-chart/README.md[specifying base and environment specific properties,role=external,window=_blank] can act as configuration properties file

Reviewer: Merge these two sentences into one:

The Configuration Store also supports a simplified way of providing configurations without mounting properties files by https://github.com/boozallen/aissemble/blob/dev/extensions/extensions-helm/aissemble-configuration-store-chart/README.md[defining properties files,role=external,window=_blank] as part of the Helm chart's values.
